### DISCUSSION TITLE
Do you think Zodios can support multiple transports in the future?

### DISCUSSION THREAD

**Discussion by thelinuxlich** [ecyrbe/zodios#166] [Q&A]
I can see Zodios workflow being very useful for service to service communication through events, streams, message queues, raw TCP, etc

**Comment by ecyrbe** [comment]
The same kind of technology that zodios is using can be indeed applied everywhere.
But, i don't have the resources (time) to do and maintain a so whide ecosystem. So zodios will be dedicated to REST over HTTP for the forseeing future.   
- If the question is, can we make full end to end typesafety on any transport? The answer is, yes, for sure.
- If the question is, can we do REST on any transport? That's a harder question to answer.
Realistically, even if REST is pretty abstract and can be used on any transport, in practice 99,99% of it is used over HTTP. Doing REST for streams, event, message queue would certainly be a challenging effort. You'll need to design what are resources in such transports, how to identify them (equivalent of URI), how to manipulate them (equivalent of HTTP verbs). This is a work for research lab PHDs. I'm  not one.

**Reply by thelinuxlich** [reply]
I don't think Zodios aims to follow purist REST definition a la Roy Fielding's essay. In fact, Roy Fielding would consider Zodios pretty much RPC: https://roy.gbiv.com/untangled/2008/rest-apis-must-be-hypertext-driven

**Reply by thelinuxlich** [reply]
I think a good middle-ground would be Zodios generalizing the transport, and supporting HTTP as the default, which injects path, parameters and method on endpoints config. Other transports would inject different required attributes.
